optional_plugins/varianter_cit/avocado_varianter_cit/__init__.py

```diff
@@ -0,0 +1,331 @@
+        Return human readable representation
+
+        The summary/variants accepts verbosity where 0 means silent and
+        maximum is up to the plugin.
+
+        :param summary: How verbose summary to output (int)
+        :param variants: How verbose list of variants to output (int)
+        :param kwargs: Other free-form arguments
+        :rtype: str
+        """
+        if not self.variants:
+            return ""
+
+        out = []
+        verbose = variants > 1
+        out.append("CIT Variants (%i):" % len(self))
+        for variant in self:
+            out.append('%sVariant %s:    %s' % ('\n' if verbose else '',
+                                                variant["variant_id"],
+                                                self.parameter_path))
+            if not verbose:
+                continue
+            env = set()
+            for node in variant["variant"]:
+                for key, value in iteritems(node.environment):
+                    origin = node.environment.origin[key].path
+                    env.add(("%s:%s" % (origin, key), str(value)))
+            if not env:
+                return out
+            fmt = '    %%-%ds => %%s' % max([len(_[0]) for _ in env])
+            for record in sorted(env):
+                out.append(fmt % record)
+
+        return "\n".join(out)
+
+
+class Cit(object):
+
+    MATRIX_ROW_SIZE = 20
+    MAX_INTERACTIONS = 15
+
+    hash_table = {}
+
+    def __init__(self, parameters, constraints, order):
+        self.parameters = parameters
+        self.data = [len(param[1]) for param in self.parameters]
+        # Constraints are not implemented in this example
+        self.constraints = constraints
+        self.order = order
+
+    def combine(self):
+        """
+        Computes the combination of parameters
+
+        :returns: headers (list of parameters keys) and combinations (list of
+                  dictionaries. Each dictionary represents a combination of
+                  parameters.
+        :rtype: tuple
+        """
+        self.create_interaction_hash_table()
+        final_list = self.create_final_list()
+
+        headers = [item[0] for item in self.parameters]
+        result = [[self.parameters[i][1][combination[i]] for i in range(len(combination))]for combination in final_list]
+        combinations = []
+        for combination in result:
+            combinations.append(dict(zip(headers, combination)))
+
+        return headers, combinations
+
+    def create_interaction_hash_table(self):
+        for c in itertools.combinations(range(len(self.data)), self.order):
+            self.hash_table[':'.join(map(str, c))] = self.get_interaction(c)
+
+    def create_final_list(self):
+        final_list = []
+        while len(self.hash_table) != 0:
+            interactions = 0
+            previous_test_case = []
+            previous_remove_list = {}
+            previous_weight = 0
+
+            while interactions < self.MAX_INTERACTIONS:
```

Comment:
I think this should be `iterations`.

Reply:
Yes, I also believe that's the most appropriate name.